app/contributing/index.md

```diff
@@ -0,0 +1,680 @@
+<!--vale off-->
+```
+{% raw %}{% kong_config_table %}
+config:
+  - name: proxy_server
+  - name: proxy_server_ssl_verify
+  - name: cluster_use_proxy
+  - name: lua_ssl_trusted_certificate
+{% endkong_config_table %}{% endraw %}
+```
+<!--vale on-->
+
+Rendered output:
+<!--vale off-->
+{% kong_config_table %}
+config:
+  - name: proxy_server
+  - name: proxy_server_ssl_verify
+  - name: cluster_use_proxy
+  - name: lua_ssl_trusted_certificate
+{% endkong_config_table %}
+<!--vale on-->
+
+### Tab groups
+
+You can use `navtabs` to create tabs with different labels.
+
+Code example:
+<!--vale off-->
+```
+{% raw %}{% navtabs "deploy-a-license" %}
+{% navtab "Admin API" %}
+
+text
+
+{% endnavtab %}
+{% navtab "license.json" %}
+
+text
+
+{% endnavtab %}
+{% endnavtabs %}{% endraw %}
+```
+<!--vale on-->
+
+Rendered output:
+<!--vale off-->
+{% navtabs "deploy-a-license" %}
+{% navtab "Admin API" %}
+
+text
+
+{% endnavtab %}
+{% navtab "license.json" %}
+
+text
+
+{% endnavtab %}
+{% endnavtabs %}
+<!--vale on-->
+
+### {{site.konnect_short_name}} roles
+
+Creates a table with the {{site.konnect_short_name}} roles and descriptions.
+<!--vale off-->
+```
+{% raw %}{% konnect_roles_table %}
+schema: control_planes
+{% endkonnect_roles_table %}{% endraw %}
+```
+
+### Version badge
+
+Displays which version a feature was introduced in. You can use this in headers, tables, and inline text.
+
+Code example:
+<!--vale off-->
+```
+{% raw %}{% new_in 3.8 %} New feature{% endraw %}
+```
+<!--vale on-->
+
+Rendered output:
+<!--vale off-->
+{% new_in 3.8 %} New feature
+<!--vale on-->
+
+### No copy code
+
+Prevents users from copying code in a code block.
+<!--vale off-->
+```
+{% raw %}{:.no-copy-code}{% endraw %}
+```
+<!--vale on-->
+
+### Render code block/text per deployment type
+
+Displays the code block/text only if the specified deployment type is selected.
+<!--vale off-->
+```
+{% raw %}{: data-deployment-topology="konnect" }
+{: data-deployment-topology="on-prem" }{% endraw %}
+```
+<!--vale on-->
+
+### Notes and warnings
+
+You can specify different types of notes and warnings.
+
+Code example:
+<!--vale off-->
+```
+{% raw %}{:.warning} # yellow note
+{:.info} # blue note
+{:.success} # green note
+{:.danger} # red note
+{:.neutral} # grey note
+{:.decorative} # purple note
+{:.info .no-icon} # add to any note type to remove the icon {% endraw %}
+```
+<!--vale on-->
+
+Rendered output:
+<!--vale off-->
+{:.warning} # yellow note
+{:.info} # blue note
+{:.success} # green note
+{:.danger} # red note
+{:.neutral} # grey note
+{:.decorative} # purple note
+{:.info .no-icon} # add to any note type to remove the icon
+<!--vale on-->
+
+### How to validations
+
+The how to pages typically include a validation step as the final step. This allows users to verify that the steps worked as intended. Validation steps are written in a `yaml` block to ensure consistency and that it is formatted correctly in the output.
+
+#### Rate limit check
```

Comment:
Might be useful to provide a link to a guide where each of there validations is in use, to show where it gets used in context.